external-crates/move/documentation/book/src/method-syntax.md

```diff
@@ -0,0 +1,442 @@
+`a::m::foo(x, e)`, potentially [automatically borrowing](#automatic-borrowing) `x`.
+
+### Functions in the Defining Module
+
+In a type’s defining module, the compiler will automatically create a method alias for any function
+declaration for its types when the type is the first argument in the function. For example,
+
+```move
+module a::m {
+    public struct X() has copy, drop, store;
+    public fun foo(x: &X) { ... }
+    public fun bar(flag: bool, x: &X) { ... }
+}
+```
+
+The function `foo` can be called as a method on a value of type `X`. However, not the first argument
+(and one is not created for `bool` since `bool` is not defined in that module). For example,
+
+```move
+fun example(x: a::m::X) {
+    x.foo(); // valid
+    // x.bar(true); ERROR!
+}
+```
+
+### `use fun` Aliases
+
+Like a traditional [`use`](uses.md), a `use fun` statement creates an alias local to its current
+scope. This could be for the current module or the current expression block. However, the alias is
+associated to a type.
+
+The syntax for a `use fun` statement is as follows:
+
+```move
+use fun <function> as <type>.<method alias>;
+```
+
+This creates an alias for the `<function>` which can be received by the `<type>` as
+`<method alias>`.
+
+For example
+
+```move
+module a::cup {
+    public struct Cup<T>(T) has copy, drop, store;
+
+    public fun cup_borrow<T>(c: &Cup<T>): &T {
+        &c.0
+    }
+
+    public fun cup_value<T>(c: Cup<T>): T {
+        let Cup(t) = c;
+        t
+    }
+
+    public fun cup_swap<T: drop>(c: &mut Cup<T>, t: T) {
+        c.0 = t;
+    }
+}
+```
+
+We can now create `use fun` aliases to these functions
+
+```move
+module b::example {
+    use fun a::cup::cup_borrow as Cup.borrow;
+    use fun a::cup::cup_value as Cup.value;
+    use fun a::cup::cup_swap as Cup.set;
+
+    fun example(c: &mut Cup<u64>) {
+        let _ = c.borrow(); // resolves to a::cup::cup_borrow
+        let v = c.value(); // resolves to a::cup::cup_value
+        c.set(v * 2); // resolves to a::cup::cup_swap
+    }
+}
+```
+
+Note that the `<function>` in the `use fun` does not have to be a fully resolved path, and an alias
+can be used instead, so the declarations in the above example could equivalently be written as
+
+```move
+    use a::cup::{Self, cup_swap};
+
+    use fun cup::cup_borrow as Cup.borrow;
+    use fun cup::cup_value as Cup.value;
+    use fun cup_swap as Cup.set;
+```
+
+While these examples are cute for renaming the functions in the current module, the feature is
+perhaps more useful for declaring methods on types from other modules. For example, if we wanted to
+add a new utility to `Cup` we could do so with a `use fun` alias and still use method syntax
+
+```move
+module b::example {
+
+    fun double(c: &Cup<u64>): Cup<u64> {
+        let v = c.value();
+        Cup::new(v * 2)
+    }
+
+}
+```
+
+Normally, we would be stuck having to call it as `double(&c)` since `b::example` did not define
```

Comment:
```suggestion
Normally, we would be stuck having to call it as `double(&c)` because `b::example` did not define
```
nit: 'since' has both a causal and temporal meaning. Because of this, because is usually preferred.